lib/pure/uri.nim

```diff
@@ -321,6 +321,34 @@ proc `$`*(u: Uri): string =
     result.add("#")
     result.add(u.anchor)
 
+proc reverseNotation*(hostname: string): string =
+  ## returns the `reverse domain name notation 
+  ## <https://en.wikipedia.org/wiki/Reverse_domain_name_notation>`_. 
+  ## of the given hostname.
+  ## A high performance version of this:
+  ## .. code-block::
+  ##   return hostname.split(".").reversed().join(".")
+  result = ""
+  var
+    pos: int = 0
+    buf: string = ""
+    ch: char
+  while true:
+    ch = hostname[pos]
+    if ch == '.' or pos == hostname.len:
+        if pos < hostname.len:
+            buf.insert ".", 0
+        result.insert(buf,0)
+        buf.setLen 0
+    else:
+        buf.add ch
+    if pos == hostname.len: break
+    pos.inc
+
+proc reverseNotation*(uri: Uri): string =
```

Comment:
I would prefer it if you just defined one procedure: `reverse(uri: Uri): string`.

Reply:
but then i need to parseUri the host first.
Maybe this code fits more inside net? 
The code this stems from does not use uri at all for eg.
This is the third time i needed this cause its so much better to sort etc... 
Where would you place it?